File: june-20/Frontend/src/app/models/imagemodel.ts
import { Post } from "./postmodel";

export interface Image {
  id?: string;             // Optional for creation (generated server-side)
  postId: string;

  name: string;            // Original file name
  content?: ArrayBuffer;   // Optional if not directly handled in the client
  isDeleted?: boolean;

  uploadedAt?: string;     // ISO string format if received from server
  post?: Post;              // Can be strongly typed as `Post` if needed
}

File: june-20/Backend/models/imagemodel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BlogPlatform.Models
{
    public class Image
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        public Guid PostId { get; set; }

        [Required]
        public string Name { get; set; } 
                
        [Required]
        public byte[] Content { get; set; } 

        public bool IsDeleted { get; set; } = false;

        public DateTime UploadedAt { get; set; } = DateTime.UtcNow;

        public Post Post { get; set; }
    }
}
